checks.d/ceph.py

```diff
@@ -82,20 +82,71 @@ def _extract_metrics(self, raw, tags):
             self.log.debug('Error retrieving osdperf metrics')
 
         try:
-            for osdinfo in raw['osd_pool_stats']:
-                name = osdinfo['pool_name']
-                local_tags = tags + ['ceph_pool:%s' % name]
+            # Health summary will be empty if no bad news
+            if raw['health_detail']['summary'] == []:
+                health = {'num_near_full_osds' : 0, 'num_full_osds' : 0}
+                self._publish(health, self.count, ['num_near_full_osds'], tags)
+                self._publish(health, self.count, ['num_full_osds'], tags)
+            else:
+                for osdhealth in raw['health_detail']['detail']:
+                    # Full looks like: osd.2 is full at 95%
+                    # Near full: osd.1 is near full at 94%
+                    pct = int(osdhealth[osdhealth.find('at ') + len('at '):osdhealth.find('%')])
+                    local_tags = tags + ['ceph_osd:%s' % osdhealth[:osdhealth.find(' is')].replace('.', '')]
+                    if 'near' in osdhealth:
+                        health = {'num_near_full_osds' : pct}
+                        self._publish(health, self.count, ['num_near_full_osds'], local_tags)
+                    else:
+                        health = {'num_full_osds' : pct}
+                        self._publish(health, self.count, ['num_full_osds'], local_tags)
+        except KeyError:
+            self.log.debug('Error retrieving health metrics')
+
+        for osdinfo in raw['osd_pool_stats']:
+            name = osdinfo['pool_name']
```

Comment:
This was unmodified from the original check: https://github.com/DataDog/dd-agent/blob/master/checks.d/ceph.py#L86

Not sure the benefit of using `get` in this case—we don't want to specify a default. But am willing to change if you see fit.